### PR TITLE
feat(email): migrate ACS rate-limit signal from Kafka to Azure Service Bus

### PR DESCRIPTION
**Pull request by Ahmed-Ghanam**
## Description
Migrates the `altinn.platform.service.updated` Kafka topic signal to Azure Service Bus for the email service's ACS rate-limiting (HTTP 429) flow.

When Azure Communication Services returns HTTP 429, the email service now dispatches an `EmailServiceRateLimitCommand` to the ASB queue `altinn.notifications.email.send.ratelimit`. The API listens on the same queue and delegates to the existing `IAltinnServiceUpdateService` to persist the rate-limit timeout in the database.

## Related Issue(s)
- #1370

## Verification
- [ ] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

## Summary by CodeRabbit

* **New Features**
  * Email service rate‑limit handling: configurable listener and publisher with selectable Azure Service Bus or Kafka routing, queue/topic names, and retry/cooldown policy.
  * New dispatcher abstraction to route rate‑limit events to the chosen transport; transient failures now emit structured rate‑limit events.

* **Tests**
  * Added unit and integration tests validating publish/consume flows, serialization, retry/backoff behavior, dead‑lettering, and timing/assertions.

### CODE REVIEW

File: components/email-service/test/Altinn.Notifications.Email.Tests/Email.Integrations/ServiceCollectionExtensionsTests.cs [hunk 1]
```diff
@@ -292,4 +292,95 @@ public void AddIntegrationServices_ListenerOnPublisherOff_RegistersKafkaDispatch
...
+
+        // Act
+        services.AddIntegrationServices(config);
+
+        // Assert
+        var descriptor = services.FirstOrDefault(d => d.ServiceType == typeof(IEmailServiceRateLimitDispatcher));
+        Assert.NotNull(descriptor);
+        Assert.Null(descriptor.ImplementationType);
+        Assert.NotNull(descriptor.ImplementationFactory);
+    }
```
[the remark this answers is not in the document]
Comment (Ahmed-Ghanam) [reply components/email-service/test/Altinn.Notifications.Email.Tests/Email.Integrations/ServiceCollectionExtensionsTests.cs:399]: The `RegisterEmailServiceRateLimitDispatcher` in `ServiceCollectionExtensions` gates the rate-limit publisher on `EnableWolverine` too

File: components/api/test/Altinn.Notifications.IntegrationTestsASB/appsettings.integrationtest.json [hunk 2]
```diff
@@ -45,3 +51,4 @@
     }
   }
 }
+
```
Comment (martivj) [inline components/api/test/Altinn.Notifications.IntegrationTestsASB/appsettings.integrationtest.json]: Remove the extra newline.

File: components/email-service/test/Altinn.Notifications.Email.IntegrationTestsASB/appsettings.integrationtest.json [hunk 3]
```diff
@@ -32,6 +32,8 @@
       "ScheduleDelaysMs": [ 500, 500, 500 ]
     },
     "EnableEmailSendResultPublisher": false,
-    "EmailSendResultQueueName": "altinn.notifications.email.send.result"
+    "EmailSendResultQueueName": "altinn.notifications.email.send.result",
+    "EnableEmailServiceRateLimitPublisher": false,
+    "EmailServiceRateLimitQueueName": "altinn.notifications.email.send.ratelimit"
   }
 }
```
Comment (martivj) [inline components/email-service/test/Altinn.Notifications.Email.IntegrationTestsASB/appsettings.integrationtest.json:39]: Refer to the comment in `EmailServiceRateLimitPublisherTests` for the motivation: these flags should be `true`, both `EnableEmailServiceRateLimitPublisher` and `EnableEmailSendResultPublisher` (which I must have missed in the last review of the send result PR).

Ref the `components/sms-service/test/Altinn.Notifications.Sms.IntegrationTestsASB/Tests/SmsSendResultPublisherTests.cs` implementation in this PR when the publisher flag is `true`: https://github.com/Altinn/altinn-notifications/pull/1491/changes, in this example we avoid having to use `.WithConfig`.

Moreover, and more crucially, there is no test file for the `EmailSendResultPublisher` setup. It is really out of scope for this issue regarding rate limiting, but we should consider adding the missing test file for proper coverage of the send result feature here (either a new issue or add it in this PR).

File: components/email-service/test/Altinn.Notifications.Email.IntegrationTestsASB/Tests/EmailServiceRateLimitPublisherTests.cs [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+        var emailClientMock = new Mock<IEmailServiceClient>();
+        emailClientMock
+            .Setup(e => e.SendEmail(It.IsAny<Core.Sending.Email>()))
+            .Callback<Core.Sending.Email>(_ => clientInvoked.TrySetResult())
+            .ReturnsAsync(new EmailClientErrorResponse { SendResult = nonTransientResult });
+
+        var factory = new IntegrationTestWebApplicationFactory(_fixture)
+            .WithConfig("WolverineSettings:EnableEmailServiceRateLimitPublisher", "true")
+            .ReplaceService(_ => emailClientMock.Object)
+            .Initialize();
```
Comment (martivj) [inline components/email-service/test/Altinn.Notifications.Email.IntegrationTestsASB/Tests/EmailServiceRateLimitPublisherTests.cs:142]: We prefer to set all feature flags to `true` in `appsettings.integrationtest.json`, so that this use of `.WithConfig` becomes unnecessary. It also makes all other tests more realistic when all queues are enabled in all tests. The intended use of `.WithConfig` in the `IntegrationTestsASB` suites is to support "opt out" cases instead of "opt in" to individual queues (e.g. we want to test an edge case with a publisher/listener pair turned off).

**Review by martivj — CHANGES_REQUESTED** [review]
Looks solid. The comments are more related to things we missed in the `EmailSendResultPublisher` testing.

File: components/api/test/Altinn.Notifications.IntegrationTestsASB/appsettings.integrationtest.json (hunk 2 above)
Comment (Ahmed-Ghanam) [reply components/api/test/Altinn.Notifications.IntegrationTestsASB/appsettings.integrationtest.json]: Removed

File: components/email-service/test/Altinn.Notifications.Email.IntegrationTestsASB/Tests/EmailServiceRateLimitPublisherTests.cs (hunk 4 above)
Comment (Ahmed-Ghanam) [reply components/email-service/test/Altinn.Notifications.Email.IntegrationTestsASB/Tests/EmailServiceRateLimitPublisherTests.cs:142]: Those redundant are deleted

**Review by martivj — APPROVED** [review]
Nice
